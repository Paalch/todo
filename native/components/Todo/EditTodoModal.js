import React, { Component } from 'react'
import { View, Item, Input, Text, Content, Title, Button, Form, H1, Grid, Col } from 'native-base';
import { Modal, TouchableHighlight } from 'react-native'
import DefaultHeader from '../DefaultHeader'
import { modalButtons, views } from '../../styles'


export default class EditTodoModal extends Component  {

    constructor(props) {
        super(props)

        //Sets state, tempText is the text in the textfield and text is the text from the todo object
        this.state = {
            tempText: this.props.text,
            text: this.props.text,
        }
    }

    /**
     * Button handler for save button
     */
    handleButtonSaveClick = () => {
        const { tempText } = this.state
        //Checks that the text in not empty
        if(tempText !== "") {
            this.props.onButtonSaveClick(tempText)
            //resets the state
            this.setState({
                tempText: '',
                text: ''
            })
            //Hides the modal
            this.props.toggleModal()
            //removes the old todo
            this.props.handleDelete()
        }
    }

    /**
     * Handle for button close click.
     * Resets the original text to state
     * Closes model
     */
    handleButtonCloseClick = () => {
        const {toggleModal} = this.props
        toggleModal()

    }

    render() {
        //define constants
        const { toggleModal, isOpen } = this.props
        const { tempText } = this.state

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={isOpen}
                onRequestClose={() => {toggleModal()}}>
                <DefaultHeader title={"Edit Todo"} toggleModal={toggleModal}/>
                <View style={views.flex1}>
                    <Content>
                        <Form>
                            <View style={views.flex1}>
                                <H1>Write your todo here:</H1>
                                <Item>
                                    <Input
                                        onChangeText={(tempText) => this.setState({tempText})}
                                        value={tempText}
                                        placeholde="Write your todo here"/>
                                </Item>
                            </View>
                        </Form>
                    </Content>
                </View>

                <View style={modalButtons.view}>
                    <Grid>
                        <Col>
                            <TouchableHighlight
                                onPress={this.handleButtonSaveClick}
                                style={modalButtons.save}>

                                <Text style={modalButtons.text}>
                                    SAVE
                                </Text>

                            </TouchableHighlight>
                        </Col>
                        <Col>

                            <TouchableHighlight
                                onPress={this.handleButtonCloseClick}
                                style={modalButtons.close}>

                                <Text style={modalButtons.text}>
                                    CANCEL
                                </Text>
                            </TouchableHighlight>

                        </Col>
                    </Grid>
                </View>

            </Modal>

        )
    }
}
