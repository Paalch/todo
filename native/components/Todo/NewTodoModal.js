import React, { Component } from 'react'
import { View, Label, Item, Input, Text, Content , Button, Form, Grid, Col } from 'native-base';
import { Modal, StyleSheet, TouchableHighlight } from 'react-native'
import DefaultHeader from '../DefaultHeader'
import { modalButtons } from '../../styles'


export default class NewTodoModal extends Component {

    constructor(props) {
        super(props)

        this.state = {
            text: '',

        }
    }

    /**
     * Handler for save button click.
     * Saves the given information from user as a new note and reset input fields
     * Close modal
     */
    handleButtonSaveClick = () => {
        const {text} = this.state
        const {toggleModal, onButtonSaveClick} = this.props

        onButtonSaveClick(text)
        this.clearInput()
        toggleModal()
    }

    /**
     * Handle for button close click.
     * Resets the original content to state
     * Closes model
     */
    handleButtonCloseClick = () => {
        const {toggleModal} = this.props
        this.clearInput()
        toggleModal()

    }

    /**
     * Clear title and content in state
     */
    clearInput = () => {
        this.setState({
            text: '',
        })
    }


    render() {
        const { isOpen } = this.props
        const { text } = this.state

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={isOpen}
                onRequestClose={() => {alert("NewTodoModal has been closed.")}}>

                <DefaultHeader title={"New Todo"} toggleModal={this.handleButtonCloseClick}/>
                <View style={styles.view}>
                    <Content >
                        <Form>
                            <View style={styles.view}>
                                <Item floatingLabel>
                                    <Label>
                                        Your todo
                                    </Label>
                                    <Input
                                        onChangeText={(text) => this.setState({text})}
                                        value={text}/>
                                </Item>
                            </View>

                        </Form>
                    </Content>

                </View>

                <View style={modalButtons.view}>
                    <Grid>
                        <Col>
                            <TouchableHighlight
                                onPress={this.handleButtonSaveClick}
                                style={modalButtons.save}>

                                <Text style={modalButtons.text}>
                                    ADD
                                </Text>

                            </TouchableHighlight>
                        </Col>
                        <Col>

                            <TouchableHighlight
                                onPress={this.handleButtonCloseClick}
                                style={modalButtons.close}>

                                <Text style={modalButtons.text}>
                                    CANCEL
                                </Text>
                            </TouchableHighlight>

                        </Col>
                    </Grid>
                </View>

            </Modal>
        )
    }
}

const styles = StyleSheet.create({
    view: {
        flex: 1,
        marginTop: 7,
        marginLeft: 7,
        marginRight: 7,
        backgroundColor: "#f5fcff"
    },
    addButtonView: {
        position:'absolute',
        bottom:0,
        width:'100%'
    },
    noteContent: {
        height: 200,
        textAlignVertical: 'top',
    },
    addButton: {
        backgroundColor: "#21ba45"
    },
    addText: {
        fontSize: 20
    }
})
