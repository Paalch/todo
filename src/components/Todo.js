import React, { Component } from 'react'
import { Segment, Checkbox } from 'semantic-ui-react'

class Todo extends Component {
    constructor(props) {
        super(props)

        console.log(props)

        this.state = {

        }
    }

    render() {
        const { todo, checkBoxClick } = this.props
        return(
            <Segment>
                { todo.text }
                <Checkbox onClick={checkBoxClick} defaultChecked={todo.checked}/>
            </Segment>
        )
    }
}

export default Todo