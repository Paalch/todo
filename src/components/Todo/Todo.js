import React, { Component } from 'react'
import { Segment, Checkbox, Grid, Divider } from 'semantic-ui-react'

import DeleteModal from '../DeleteModal'

class Todo extends Component {
    constructor(props) {
        super(props)

        console.log(props)

        this.state = {

        }
    }

    handleCheckBoxClick = () => {
        const { todo, checkBoxClick } = this.props
        checkBoxClick(todo)
    }

    handleDelete = () => {
        const { todo, deleteItem } = this.props
        deleteItem(todo)
    }

    render() {
        const { todo } = this.props
        return(
          <div>
            <Segment color={todo.checked ? 'green' : undefined} inverted={todo.checked} >
                <Grid
                  verticalAlign="middle"
                  width={16}>
                    <Grid.Column width={13}>
                        { todo.text }
                    </Grid.Column>
                    <Grid.Column width={1}>
                        <Checkbox onClick={this.handleCheckBoxClick} defaultChecked={todo.checked} />
                    </Grid.Column>
                    <Grid.Column width={2}>
                        <DeleteModal handleDelete={this.handleDelete} title={"Todo"}/>
                    </Grid.Column>
                </Grid>
            </Segment>
          </div>
        )
    }
}

export default Todo