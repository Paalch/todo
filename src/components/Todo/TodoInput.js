import React, { Component } from 'react'
import { Input, Button } from 'semantic-ui-react'

export default class TodoInput extends Component {
    constructor(props) {
        super(props)

        this.state = {
            text: ''
        }

        this.onChange = this.onChange.bind(this)
        this.onButtonClick = this.onButtonClick.bind(this)
    }

    onChange(e, {value}) {
        this.setState({
            text: value
        })
    }

    onButtonClick() {
        const { text } = this.state
        this.props.onButtonClick(text)
        this.setState({
            text: ''
        })
    }

    render() {
        const { text } = this.state
        return(
            <div>
                <Input action fluid placeholder='Write your Todo...' onChange={this.onChange} value={text}>
                    <input />
                    <Button icon='plus' color='green' onClick={this.onButtonClick}/>
                </Input>
            </div>
        )
    }
}