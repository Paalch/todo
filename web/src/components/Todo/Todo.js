import React, { Component } from 'react'
import { Segment, Checkbox, Grid, Button, Icon } from 'semantic-ui-react'
import moment from 'moment'
import EditToDoModal from './EditTodoModal'
import DeleteModal from '../DeleteModal'

class Todo extends Component {
    constructor (props) {
        super(props)

        // init state
        this.state = {
            editModalOpen: false,
            date: this.props.todo.date
        }
    }

    // Handles the checkbox
    handleCheckBoxClick = () => {
        const { todo, checkBoxClick } = this.props
        checkBoxClick(todo)
    }

    // Removes this todo
    handleDelete = () => {
        const { todo, deleteItem } = this.props
        deleteItem(todo)
    }

    // opens and close the editmodal by setting editModalOpen
    toggleEditModal = () => {
        this.setState({
            editModalOpen: !this.state.editModalOpen
        })
    }

    // Adds this todo to favorites
    markAsFavorite = () => {
        let { todo } = this.props
        todo.isStar = !todo.isStar
        this.props.updateToDos(todo)
    }

    render () {
        // define constants
        const { date, editModalOpen } = this.state
        const { todo, onButtonClick } = this.props

        return (
            <div>
                <Segment color={todo.checked ? 'green' : undefined} inverted={todo.checked}>
                    <Grid verticalAlign="middle">
                        <Grid.Column width={1}>
                            <Icon name={todo.isStar ? 'star' : 'empty star'} color='yellow' onClick={this.markAsFavorite}/>
                        </Grid.Column>
                        <Grid.Column width={7}>
                            { todo.text }
                        </Grid.Column>
                        <Grid.Column width={4} onClick={this.toggleViewModal}>
                            { moment(date).calendar()}
                        </Grid.Column>
                        <Grid.Column width={1}>
                            <Checkbox onClick={this.handleCheckBoxClick} defaultChecked={todo.checked} />
                        </Grid.Column>
                        <Grid.Column width={2}>
                            <Button.Group>
                                <DeleteModal handleDelete={this.handleDelete} />
                                <Button icon="edit" color="grey" onClick={this.toggleEditModal}/>
                            </Button.Group>
                        </Grid.Column>

                    </Grid>
                </Segment>

                <EditToDoModal
                    isOpen={editModalOpen}
                    onClose={this.toggleEditModal}
                    onButtonSaveClick = {onButtonClick}
                    handleDelete = {this.handleDelete}
                    date={moment(date).format('H:mm A, MMM Do YYYY')}
                    content={todo.text}
                />
            </div>
        )
    }
}

export default Todo
