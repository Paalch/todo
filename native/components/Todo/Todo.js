import React, { Component } from 'react'
import { Segment, CheckBox, Button, Icon, Content, Text, ListItem, Item, Container } from 'native-base'
import { Col, Grid, Row } from 'react-native-easy-grid';
import moment from 'moment'
import EditTodoModal from "./EditTodoModal"


export default class Todo extends Component<{}> {
    constructor(props) {
        super(props)

        //sets the state
        this.state = {
            //Boolean used to open and close modal
            editModalOpen: false,
            //Sets the date to props date
            date: this.props.todo.date,
        }
    }

    //Handels checkbox click
    handleCheckBoxClick = () => {
        const { todo, checkBoxClick } = this.props
        checkBoxClick(todo)
    }

    //Handles delete button, removes this item
    handleDelete = () => {
        const { todo, deleteItem } = this.props
        deleteItem(todo)
    }

    //Toggele the modal window
    toggleEditModal = () => {
        this.setState({
            editModalOpen: !this.state.editModalOpen
        })
    }

    //Sets this todo as a favorite
    markAsFavorite = () => {
        let { todo } = this.props
        todo.isStar = !todo.isStar
        this.props.updateToDos(todo)
    }


    render() {
        //defines constants
        const { date, editModalOpen} = this.state
        const { todo, onButtonClick} = this.props
        return(
            <ListItem style={{backgroundColor: todo.checked ? "#4BB543" : "white" }}>
                    <Content>
                        <Grid>
                            <Col size={15}>
                                <CheckBox onPress={this.handleCheckBoxClick} checked={todo.checked} />
                            </Col>
                            <Col size={85}>
                                <Row>
                                    <Text style={{fontSize: 21, paddingBottom: 2}}>{ todo.text } </Text>
                                </Row>
                                <Row>
                                    <Col size={45}>
                                        <Text>{ moment(date).calendar()}</Text>
                                    </Col>
                                    <Col size={55}>
                                        <Item>
                                            <Button onPress={this.toggleEditModal} success>
                                                <Icon name='create'/>
                                            </Button>
                                            <Button onPress={this.handleDelete} danger>
                                                <Icon name='trash'/>
                                            </Button>
                                            <Button onPress={this.markAsFavorite} warning>
                                                <Icon name={todo.isStar ? 'md-star' : 'star'}/>
                                            </Button>
                                        </Item>
                                    </Col>
                                </Row>
                            </Col>
                          </Grid>
                        <EditTodoModal toggleModal={this.toggleEditModal}
                                       isOpen={editModalOpen}
                                       onButtonSaveClick={onButtonClick}
                                       handleDelete = {this.handleDelete}
                                       content = {todo.text} />
                </Content>
            </ListItem>
        )
    }
}
