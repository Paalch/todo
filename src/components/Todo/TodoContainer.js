import React, { Component } from 'react'
import { Container, Divider } from 'semantic-ui-react'

import Todo from './Todo'
import TodoInput from './TodoInput'
import Navbar from '../Navbar'
import Footer from "../Footer";

export default class TodoContainer extends Component {
    constructor(props) {
        super(props)

        this.state = {
            todos: []
        }
    }

    componentWillMount = () => {
        //localStorage.clear()
        let localTodos = JSON.parse(localStorage.getItem('todos'))
        this.setState({
            todos: localTodos || []
        })
    }

    checkBoxClick = (todo) => {
        const { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos[i].checked = !todos[i].checked
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
        
    }

    onButtonClick = (text) => {
        const { todos } = this.state
        let todo = {
            text: text,
            date: new Date(),
            checked: false
        }
        todos.push(todo)
        this.updateState({
            todos: todos
        })
    }

    deleteItem = (todo) => {
        let { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos.splice(i, 1)
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

    updateLocalStorage = () => {
        const { todos } = this.state
        localStorage.setItem('todos', JSON.stringify(todos))
    }

    updateState = (state) => {
        this.setState(state, () => {
            this.updateLocalStorage()
        })
    }

    render() {
        const { todos } = this.state
        return(
            <div>
                <Navbar />
                <Container text>
                    <Divider hidden />
                    <TodoInput onButtonClick={this.onButtonClick}/>
                    <Divider hidden/>
                    { todos.map((todo) => <Todo key={todo.date} todo={todo} checkBoxClick={this.checkBoxClick} deleteItem={this.deleteItem}/>) }
                </Container>
                <Divider hidden />
                <Footer/>
            </div>
        )
    }
} 