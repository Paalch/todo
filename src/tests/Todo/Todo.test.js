import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Todo from '../../components/Todo/Todo';
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'
import moment from 'moment'


/*describe('About component', () => {
  it('', () => {

  })
});*/

Enzyme.configure({ adapter: new Adapter() });

test('Todo component should render as expected', () => {
  const todo = {date: moment()};
  const component = shallow(<Todo todo={todo}/>);
  const tree = toJson(component);
  console.log(tree);
  expect(tree).toMatchSnapshot();
});

