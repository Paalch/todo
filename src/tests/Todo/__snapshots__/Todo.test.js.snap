// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should render as expected 1`] = `
<div>
  <Segment>
    <Grid
      verticalAlign="middle"
    >
      <GridColumn
        width={1}
      >
        <Icon
          as="i"
          color="yellow"
          name="empty star"
          onClick={[Function]}
        />
      </GridColumn>
      <GridColumn
        width={7}
      />
      <GridColumn
        width={4}
      >
        06/20/2012
      </GridColumn>
      <GridColumn
        width={1}
      >
        <Checkbox
          onClick={[Function]}
          type="checkbox"
        />
      </GridColumn>
      <GridColumn
        width={2}
      >
        <ButtonGroup>
          <DeleteModal
            handleDelete={[Function]}
          />
          <Button
            as="button"
            color="grey"
            icon="edit"
            onClick={[Function]}
          />
        </ButtonGroup>
      </GridColumn>
    </Grid>
  </Segment>
  <EditToDoModal
    date="0:00 AM, Jun 20th 2012"
    handleDelete={[Function]}
    isOpen={false}
    onClose={[Function]}
  />
</div>
`;
